### PR TITLE
Add USE_CIRUN killswitch to disable cirun VMs if needed

### PR DESCRIPTION
**Pull request by kendalharland**
Use `vars.USE_CIRUN` to disable arm64 jobs that run on our Azure VMs.

This can be overridden at the org or repo level by setting the org or repo value of `USE_CIRUN='false'`.

## Test
- Kill switch triggered: https://github.com/thebrowsercompany/swift-build/actions/runs/8932515939
- Kill switch not triggered: https://github.com/thebrowsercompany/swift-build/actions/runs/8932557779

### CODE REVIEW

File: .github/workflows/release-swift-toolchain-binary-sizes.yml [hunk 1]
```diff
@@ -47,11 +85,7 @@ jobs:
 
     strategy:
       matrix:
-        include:
-          - arch: amd64
-            os: windows-latest
-          - arch: arm64
-            os: cirun-win11-23h2-pro-arm64-16-2024-04-22
```
Comment (maruel) [inline .github/workflows/release-swift-toolchain-binary-sizes.yml:54]: Why not use the same form as used in the other repo? Does it not work?

Comment (kendalharland) [reply .github/workflows/release-swift-toolchain-binary-sizes.yml:54]: No because in those repos we're not specifying the runners using a matrix - we're just choosing to run on either the windows-latest or the cirun vms but not both, as we are here. :/ Github doesn't allow conditional matrix entries, and the semantics of `jobs.<job-id>.if` and `jobs.<job-id>.strategy.matrix.exclude` don't provide a simpler solution for matrix exclusion.

Comment (maruel) [reply .github/workflows/release-swift-toolchain-binary-sizes.yml:54]: I mean:
```
- arch: arm64
  os: "${{ vars.USE_CIRUN == 'true' && format('cirun-win11-23h2-pro-arm64-64-2024-04-22--{0}', github.run_id) || 'windows-latest' }}
```

This doesn't work? That's too bad if it's the case.

Comment (kendalharland) [reply .github/workflows/release-swift-toolchain-binary-sizes.yml:54]: Ah I see. Yeah that won't work because it'd attempt to install the arm64 Swift toolchain on the x64 windows-latest image which doesn't work from my last attempts... but now I'm second guessing myself. Let me try again.

Comment (maruel) [reply .github/workflows/release-swift-toolchain-binary-sizes.yml:54]: Ah ok it makes sense. Then I'd recommend to use https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idif

Comment (kendalharland) [reply .github/workflows/release-swift-toolchain-binary-sizes.yml:54]: Sadly, that doesn't work because of:
> Note: The jobs.<job_id>.if condition is evaluated before [jobs.><job_id>.strategy.matrix](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix) is applied.

Which means we can't write something like:

```
if: ${{ matrix.is_cirun == "false" || vars.USE_CIRUN }}
```

Comment (maruel) [reply .github/workflows/release-swift-toolchain-binary-sizes.yml:54]: sad panda

**Review by maruel — APPROVED** [review]
